### ISSUE TITLE
asm_mnd.txt does not exist or does not contain any reads.

### ISSUE THREAD

**Issue by naktang1** [aidenlab/3d-dna#107]
Hello! I've been trying to run the run-asm-pipeline.sh with the following command

/tarafs/data/project/proj0156-DRQTL/software/3d-dna/run-asm-pipeline.sh -m diploid kongkang_ct_editname.fasta /tarafs/data/project/proj0156-DRQTL/test_3DNA/aligned/merged_nodups.txt

in a next few minute, I got an error message

/tarafs/data/project/proj0156-DRQTL/software/3d-dna/run-asm-pipeline.sh -m diploid kongkang_ct_editname.fasta /tarafs/data/project/proj0156-DRQTL/test_3DNA/aligned/merged_nodups.txt
version: 180922
 -m|--mode flag was triggered. Running in diploid mode.
/tarafs/data/project/proj0156-DRQTL/software/3d-dna/run-asm-pipeline.sh: line 656: [: too many arguments
###############
Starting iterating scaffolding with editing:
...starting round 0 of scaffolding:
:) -p flag was triggered. Running LIGer with GNU Parallel support parameter set to true.
:) -s flag was triggered, starting calculations with 15000 threshold starting contig/scaffold size
:) -q flag was triggered, starting calculations with 1 threshold mapping quality
...Using cprops file: kongkang_ct_editname.0.cprops
...Using merged_nodups file: kongkang_ct_editname.mnd.0.txt
...Scaffolding all scaffolds and contigs greater or equal to 15000 bp.
...Starting iteration # 1
:) DONE!
...visualizing round 0 results:
:) -p flag was triggered. Running with GNU Parallel support parameter set to true.
:) -q flag was triggered, starting calculations for 1 threshold mapping quality
:) -i flag was triggered, building mapq without
:) -c flag was triggered, will remove temporary files after completion
...Remapping contact data from the original contig set to assembly
...Building track files
...Building the hic file
temp.kongkang_ct_editname.0.asm_mnd.txt does not exist or does not contain any reads.
...detecting misjoins in round 0 assembly:
:) -p flag was triggered. Running with GNU Parallel support parameter set to true.
:) -c flag was triggered, starting calculations with 5% saturation level
:) -w flag was triggered, performing cursory search for mismatches at 25000 resolution
:) -d flag was triggered, depletion score will be averaged across a region bounded by 125000 superdiagonal
:) -k flag was triggered, starting calculations with 55% depletion as mismatch threshold
:) -n flag was triggered, performing mismatch region thinning at 1000 resolution
...Dumping 25000 resolution matrix
Could not read hic file: null
:( Juicebox dump is empty! Perhaps something is wrong with the hic file or the requested resolution is too high. Exiting!
:) -w flag was triggered, performing cursory search for repeat at 25000 resolution
:) -t flag was triggered, flagging regions with coverage higher than 2
...Dumping 25000 resolution coverage track
Could not read hic file: null
:( Juicebox coverage dump is empty! Perhaps something is wrong with the hic file or the requested resolution is too high. Exiting!
mv: cannot stat โ€depletion_score_wide.wigโ€: No such file or directory
mv: cannot stat โ€depletion_score_narrow.wigโ€: No such file or directory
mv: cannot stat โ€mismatch_wide.bedโ€: No such file or directory
mv: cannot stat โ€mismatch_narrow.bedโ€: No such file or directory
mv: cannot stat โ€coverage_wide.wigโ€: No such file or directory
mv: cannot stat โ€repeats_wide.bedโ€: No such file or directory
cat: mismatch_narrow.at.step.0.bed: No such file or directory
cat: repeats_wide.at.step.0.bed: No such file or directory
...-p flag was triggered. Running with GNU Parallel support parameter set to true.
...-r flag was triggered, output will be labeled as *.1.*
...applying edits to cprops file
...applying edits to mnd file
...starting round 1 of scaffolding:
:) -p flag was triggered. Running LIGer with GNU Parallel support parameter set to true.
:) -s flag was triggered, starting calculations with 15000 threshold starting contig/scaffold size
:) -q flag was triggered, starting calculations with 1 threshold mapping quality
...Using cprops file: kongkang_ct_editname.1.cprops
...Using merged_nodups file: kongkang_ct_editname.mnd.1.txt
...Scaffolding all scaffolds and contigs greater or equal to 15000 bp.
...Starting iteration # 1
:) DONE!
...visualizing round 1 results:
:) -p flag was triggered. Running with GNU Parallel support parameter set to true.
:) -q flag was triggered, starting calculations for 1 threshold mapping quality
:) -i flag was triggered, building mapq without
:) -c flag was triggered, will remove temporary files after completion
...Remapping contact data from the original contig set to assembly
...Building track files
...Building the hic file
temp.kongkang_ct_editname.1.asm_mnd.txt does not exist or does not contain any reads.
...detecting misjoins in round 1 assembly:
:) -p flag was triggered. Running with GNU Parallel support parameter set to true.
:) -c flag was triggered, starting calculations with 5% saturation level
:) -w flag was triggered, performing cursory search for mismatches at 25000 resolution
:) -d flag was triggered, depletion score will be averaged across a region bounded by 125000 superdiagonal
:) -k flag was triggered, starting calculations with 55% depletion as mismatch threshold
:) -n flag was triggered, performing mismatch region thinning at 1000 resolution
...Dumping 25000 resolution matrix
Could not read hic file: null
:( Juicebox dump is empty! Perhaps something is wrong with the hic file or the requested resolution is too high. Exiting!
:) -w flag was triggered, performing cursory search for repeat at 25000 resolution
:) -t flag was triggered, flagging regions with coverage higher than 2
...Dumping 25000 resolution coverage track
Could not read hic file: null
:( Juicebox coverage dump is empty! Perhaps something is wrong with the hic file or the requested resolution is too high. Exiting!
mv: cannot stat โ€depletion_score_wide.wigโ€: No such file or directory
mv: cannot stat โ€depletion_score_narrow.wigโ€: No such file or directory
mv: cannot stat โ€mismatch_wide.bedโ€: No such file or directory
mv: cannot stat โ€mismatch_narrow.bedโ€: No such file or directory
mv: cannot stat โ€coverage_wide.wigโ€: No such file or directory
mv: cannot stat โ€repeats_wide.bedโ€: No such file or directory
cat: mismatch_narrow.at.step.1.bed: No such file or directory
cat: repeats_wide.at.step.1.bed: No such file or directory
...-p flag was triggered. Running with GNU Parallel support parameter set to true.
...-r flag was triggered, output will be labeled as *.2.*
...applying edits to cprops file
...applying edits to mnd file
...starting round 2 of scaffolding:
:) -p flag was triggered. Running LIGer with GNU Parallel support parameter set to true.
:) -s flag was triggered, starting calculations with 15000 threshold starting contig/scaffold size
:) -q flag was triggered, starting calculations with 1 threshold mapping quality
...Using cprops file: kongkang_ct_editname.2.cprops
...Using merged_nodups file: kongkang_ct_editname.mnd.2.txt
...Scaffolding all scaffolds and contigs greater or equal to 15000 bp.
...Starting iteration # 1
:) DONE!
...visualizing round 2 results:
:) -p flag was triggered. Running with GNU Parallel support parameter set to true.
:) -q flag was triggered, starting calculations for 1 threshold mapping quality
:) -i flag was triggered, building mapq without
:) -c flag was triggered, will remove temporary files after completion
...Remapping contact data from the original contig set to assembly
...Building track files
...Building the hic file
temp.kongkang_ct_editname.2.asm_mnd.txt does not exist or does not contain any reads.
:( No resolved files are found. Please rerun the pipeline to include the scaffold segment. Exiting!

It seems like some file was not create but I don't understand why it happens. Do you have any advice to solve this problem?

**Comment by dudcha** [comment]
Please run with bash. My guess this is not your default shell. -Olga

**Comment by naktang1** [comment]
Unfortunately, I try to run with bash script but I got the same error

**Comment by dudcha** [comment]
My guess that you have spaces somewhere in the path yielding you this and probably downstream problems:
/tarafs/data/project/proj0156-DRQTL/software/3d-dna/run-asm-pipeline.sh: line 656: [: too many arguments
Try putting your files somewhere without spaces. -Olga

**Comment by naktang1** [comment]
Dear Olga

Thank you for your suggestion. I try to put all of my files in a new
directory without space. However, I also got an error.

Could not read hic file: null
:( Juicebox dump is empty! Perhaps something is wrong with the hic file or
the requested resolution is too high. Exiting!


Chaiwat

ในวันที่ ศ. 9 เม.ย. 2021 เวลา 00:43 dudcha ***@***.***>
เขียนว่า:

**Comment by dudcha** [comment]
Please be a bit more specific. Is this the first error message? Do you have correct permissions set for the fasta and mnd file? Can you manually read them?

**Comment by naktang1** [comment]
Dear Olga

After I rechecked the error message then I found that the first error
message is still the same.
"asm_mnd.txt does not exist or does not contain any reads"

Moreover, I can edit both fasta and mnd file manually.

I provide you a header of both fasta and mnd file

fasta file

>contig1
TTCTCTTGCCTCAAAGTCTCCTCTACTCTTGAGCTTAAAAAGAGTTGAGCAATTATGAGAGGCTTATAAAGGCAAGGTCT
TTAAAGTCTCCTAGTGAGAAAATGTTTGCCTCAAACCTTTCTAAAATACTCACTAAGACCTTATGTACAATACTTCTATC
TAGGGAGTCTTCTCCTAAAACCTTGATTTGGTTAACAATCTTCAAAGGTTTATCTACATACTCCCTTAAGGTTATTACCT
CCTTTATCCTTAAGATTTTAAACTCTTTTAAGGTTAAGGAGTTACATGTTTTTAATTTTCTCATTACTAGAAAACTCCTT
CTTCAACCTTTCTTATACATCCCTGGTAGTTTCATTGGTCATACTCCTATTGAAAATGGTCTTTGACATCACTGAGTGAA
TGTAAGTCAAGGCTAGGTACTGTTTAGTTGTTTACTCATTATGTTGTTTCATTTGTGCTAGAGAAGATAAGTAGGCAAGG
GAGGTGGTTCAATACCTTTTTCAATGGCTTCCCACAAGTCATATGCTTTAACAAAGGTTTTTGTTATTATTATCCATTCA
AGATAATTTTTCTCAATAAATTTTTAAAAAGAACGAAAAGAAATTAAAGGAGACCATATAAAAAAAGGTGTTTGAGAGTT
TTAGCAAGATTTTATTAGAATTTTGGTTTAAGAAGGAATGGTTAATTTCCTTAGTGATTATTTCTCAAGTTTGTAGGATA

mnd file

0 contig1 710 0 0 contig1 925 0 18 150M
TTTGTAGGATATTACTAACTATGATATTAGTTTGTTGGATTTTTGTTACTAAGACAAGGAATTTGGCATATCAGTTATTGGTTTGCTACTAATGTGTATGTGTGTGCTGGTTTGCAAGAGGGTTGTAATAATTTTCTATTTTTGCTCAAA
49 13S92M1D45M
GATGGACGAACCTCAAAAAAGGCTTACATAATGGATTTAGAGGTTCTCACATATAAAAGTAGTTTTACATTAGTTTAAACACCAAAATTAGTTTTACAGTTCCCTAAGTTGGTTGTATTAATTAACATAATTATTATAACTACTAATTAT
GWNJ-0850:870:GW2101183646th:1:2113:14286:28699
GWNJ-0850:870:GW2101183646th:1:2113:14286:28699
0 contig1 24 0 16 contig1 300 0 7 150M
TACTCTTGAGCTTAAAAAGAGTTGAGCAATTATGAGAGGCTTATAAAGGCAAGGTCTTTAAAGTCTCCTAGTGAGAAAATTTTTGCCTTAAACCTTTCTAAAATACTCACTAAGACCTTATGTACAATACTTCTATCTAGGGAGTCTTCT
52 94M56S
TACTTCTATCTAGGGAGTCTTCTCCTAAAACCTTGATTTGGTTAACAATCTTCAAAGGTTTATCTACATACTCCCTTAAGGTTATTACCTCCTTAGGTTCGTCCATCGATCGATGGACGAACCTGGCAATAACCTGAGATGCAGTAGACC
GWNJ-0850:870:GW2101183646th:1:2113:10998:57038
GWNJ-0850:870:GW2101183646th:1:2113:10998:57038

The full error message is shown below

Starting iterating scaffolding with editing:
...starting round 0 of scaffolding:
:) -p flag was triggered. Running LIGer with GNU Parallel support parameter
set to true.
:) -s flag was triggered, starting calculations with 15000 threshold
starting contig/scaffold size
:) -q flag was triggered, starting calculations with 1 threshold mapping
quality
...Using cprops file: kongkang_ct_editname_3DNA.0.cprops
...Using merged_nodups file: kongkang_ct_editname_3DNA.mnd.0.txt
...Scaffolding all scaffolds and contigs greater or equal to 15000 bp.
...Starting iteration # 1
:) DONE!
...visualizing round 0 results:
:) -p flag was triggered. Running with GNU Parallel support parameter set
to true.
:) -q flag was triggered, starting calculations for 1 threshold mapping
quality
:) -i flag was triggered, building mapq without
:) -c flag was triggered, will remove temporary files after completion
...Remapping contact data from the original contig set to assembly
...Building track files
...Building the hic file
temp.kongkang_ct_editname_3DNA.0.asm_mnd.txt does not exist or does not
contain any reads.
...detecting misjoins in round 0 assembly:
:) -p flag was triggered. Running with GNU Parallel support parameter set
to true.
:) -c flag was triggered, starting calculations with 5% saturation level
:) -w flag was triggered, performing cursory search for mismatches at 25000
resolution
:) -d flag was triggered, depletion score will be averaged across a region
bounded by 125000 superdiagonal
:) -k flag was triggered, starting calculations with 55% depletion as
mismatch threshold
:) -n flag was triggered, performing mismatch region thinning at 1000
resolution
...Dumping 25000 resolution matrix
Could not read hic file: null
:( Juicebox dump is empty! Perhaps something is wrong with the hic file or
the requested resolution is too high. Exiting!
:) -w flag was triggered, performing cursory search for repeat at 25000
resolution
:) -t flag was triggered, flagging regions with coverage higher than 2
...Dumping 25000 resolution coverage track
Could not read hic file: null
:( Juicebox coverage dump is empty! Perhaps something is wrong with the hic
file or the requested resolution is too high. Exiting!
mv: cannot stat ‘depletion_score_wide.wig’: No such file or directory
mv: cannot stat ‘depletion_score_narrow.wig’: No such file or directory
mv: cannot stat ‘mismatch_wide.bed’: No such file or directory
mv: cannot stat ‘mismatch_narrow.bed’: No such file or directory
mv: cannot stat ‘coverage_wide.wig’: No such file or directory
mv: cannot stat ‘repeats_wide.bed’: No such file or directory
cat: mismatch_narrow.at.step.0.bed: No such file or directory
cat: repeats_wide.at.step.0.bed: No such file or directory



ในวันที่ ศ. 9 เม.ย. 2021 เวลา 09:47 dudcha ***@***.***>
เขียนว่า:

> Please be a bit more specific. Is this the first error message? Do you
> have correct permissions set for the fasta and mnd file? Can you manually
> read them?
>
